sql/analyzer/resolve_subqueries.go

```diff
@@ -50,9 +50,31 @@ func finalizeSubqueries(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope,
 // finalizeSubqueriesHelper finalizes all subqueries and subquery expressions,
 // fixing parent scopes before recursing into child nodes.
 func finalizeSubqueriesHelper(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
-	return transform.Node(node, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+	var joinParent *plan.JoinNode
+	var selFunc transform.SelectorFunc = func(c transform.Context) bool {
+		if jp, ok := c.Node.(*plan.JoinNode); ok {
+			joinParent = jp
+		}
+		return true
+	}
+
+	var conFunc transform.CtxFunc = func(c transform.Context) (sql.Node, transform.TreeIdentity, error) {
+		n := c.Node
 		if sqa, ok := n.(*plan.SubqueryAlias); ok {
-			newSqa, same2, err := analyzeSubqueryAlias(ctx, a, sqa, scope, sel, true)
+			var newSqa sql.Node
+			var same2 transform.TreeIdentity
+			var err error
+			if sqa.OuterScopeVisibility && joinParent != nil && !joinParent.Op.IsLeftOuter() {
+				if stripChild, ok := joinParent.Right().(*plan.StripRowNode); ok && stripChild.Child == sqa {
+					subScope := scope.newScopeInJoin(joinParent.Children()[0])
```

Comment:
We should add a comment that this only fixes 1-degree of join nesting, and that a proper fix would re-index the entire tree at the end of analysis. If the subquery has more than 1 sibling, or a sibling higher in the join tree, we currently do not detect it. We probably don't need tests for those edge cases yet. Just gotta hope no one hits them soon.